qt/scientific_interfaces/Inelastic/Analysis/FitDataModel.cpp

```diff
@@ -186,25 +186,24 @@ void IndirectFitDataModel::addWorkspace(const std::string &workspaceName, const
   addWorkspace(ws, spectra);
 }
 
-void IndirectFitDataModel::addWorkspace(Mantid::API::MatrixWorkspace_sptr workspace,
-                                        const FunctionModelSpectra &spectra) {
+void FitDataModel::addWorkspace(Mantid::API::MatrixWorkspace_sptr workspace, const FunctionModelSpectra &spectra) {
   if (!m_fittingData->empty()) {
-    for (auto &fitData : *m_fittingData) {
-      if (equivalentWorkspaces(workspace, fitData.workspace())) {
-        fitData.combine(IndirectFitData(workspace, spectra));
-        return;
-      }
+    auto it = std::find_if((*m_fittingData).begin(), (*m_fittingData).end(), [&](FitData const &fitData) {
+      return equivalentWorkspaces(workspace, fitData.workspace());
+    });
+    if (it != (*m_fittingData).end()) {
+      (*it).combine(FitData(workspace, spectra));
```

Comment:
In the old code, there has been a return right after `fitData.combine(IndirectFitData(workspace, spectra));` causing not to call `addNewWorkspace(workspace, spectra);` 
But in the refactored code the return is missing- just noted in case it is not a mistake.

Reply:
Thanks for spotting this!